## Lab | Inferential statistics - ANOVA

**Part 1**

In this activity, we will look at another example. Your task is to understand the problem and write down all the steps to set up ANOVA. After the next lesson, we will ask you to solve this problem using Python. Here are the steps that you would need to work on: - Null hypothesis - Alternate hypothesis - Level of significance - Test statistic - P-value - F table

In [2]:
# Importing necessary liabraries
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

import statsmodels.api as sm
from statsmodels.formula.api import ols
from scipy.stats import ttest_ind
import statsmodels.api as sm

In [3]:
# Reading a file
data = pd.read_excel('anova_lab_data.xlsx')
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [4]:
# Checking data information
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Power         15 non-null     object 
 1   Etching Rate  15 non-null     float64
dtypes: float64(1), object(1)
memory usage: 368.0+ bytes


In [5]:
# Checking data columns
data.columns

Index(['Power ', 'Etching Rate'], dtype='object')

In [6]:
# formatting header names
data.columns = [data.columns[i].lower().replace(' ','_') for i in range(len(data.columns))]

In [7]:
data.columns

Index(['power_', 'etching_rate'], dtype='object')

In [8]:
data.rename(columns={'power_':'power'}, inplace=True)

In [9]:
data.columns

Index(['power', 'etching_rate'], dtype='object')

In [11]:
data.describe().T

count      mean       std   min    25%   50%    75%  max
etching_rate   15.0  6.782667  1.228643  5.43  5.845  6.24  7.725  9.2

In [16]:
data['power'].value_counts()

160 W    5
180 W    5
200 W    5
Name: power, dtype: int64

In [17]:
data.groupby('power').agg(np.mean)

etching_rate
power              
160 W         5.792
180 W         6.238
200 W         8.318

 - **State the null hypothesis**
(H0): mean(160 w) = mean(180 w) = mean(200 w)

 - **State the alternate hypothesis**
(H1): At least one of the mean etching rates is not similiar with other two

 - **What is the significance level**
α = 0.05 

 - **What are the degrees of freedom of model, error terms, and total DoF**
     - The model DoF = no of groups - 1 = 3-1 = 2

     - The error terms DoF = the number of observations - the number of groups = 15-3 = 12

     - The total DoF = the number of observations - 1 = 15-1= 14

**Part 2**
 - In this section, use Python to conduct ANOVA.
 - What conclusions can you draw from the experiment and why?

In [18]:
model = ols('etching_rate ~ C(power)', data=data).fit()
sm.stats.anova_lm(model)

df     sum_sq   mean_sq          F    PR(>F)
C(power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN

**Considering a significance level of 0.05, the p-value 0.000008 is below, so we reject the null hypothesis.**